testsuite/tests/kuadrant/gateway/conftest.py

```diff
@@ -0,0 +1,40 @@
+"""Conftest for gateway tests"""
+import pytest
+
+from testsuite.httpx.auth import HttpxOidcClientAuth
+from testsuite.openshift.objects.auth_config.auth_policy import AuthPolicy
+
+
+@pytest.fixture(scope="module")
+def kuadrant(kuadrant):
+    """Skip if not running on Kuadrant"""
+    if not kuadrant:
+        pytest.skip("Gateway tests are only for Kuadrant")
+    return kuadrant
+
+
+@pytest.fixture(scope="module", autouse=True)
+def gateway_wait_for_ready(gateway):
+    """Waits for gateway to be ready"""
+    gateway.wait_for_ready()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def authorization_wait_for_ready(gateway_wait_for_ready, commit, route, authorization):
+    # pylint: disable=unused-argument
+    """Waits for authorization to be ready"""
+    authorization.wait_for_ready()
```

Comment:
You can add it to the commit fixture in on of the upper contest, so every test waits for ready